compiler/rustc_traits/src/implied_outlives_bounds.rs

```diff
@@ -117,7 +114,7 @@ fn compute_implied_outlives_bounds<'tcx>(
 
     // Ensure that those obligations that we had to solve
     // get solved *here*.
-    match fulfill_cx.select_all_or_error(infcx).as_slice() {
+    match ocx.select_all_or_error().as_slice() {
         [] => (),
         _ => return Err(NoSolution),
     }
```

Comment:
What does calling `select_all_or_error` here do to the `select_*` calls that happen in `make_query_response`, since we're inside of a canonical query here?

Reply:
(Maybe it does nothing -- just curious, since we're passing down the `ObligationCtxt` from the canonical query instead of creating a trait engine inline, it might change behavior -- but maybe not!)

Reply:
it is necessary to constrain inference variables which we use further down when computing the implied bounds.